src/vs/workbench/common/editor/editorInput.ts

```diff
@@ -106,8 +111,12 @@ export abstract class EditorInput extends Disposable implements IEditorInput {
 		return this;
 	}
 
-	matches(otherInput: unknown): boolean {
-		return this === otherInput;
+	matches(otherInput: IEditorInput | IResourceEditorInputType): boolean {
+		if (otherInput instanceof EditorInput) {
+			return this === otherInput;
+		} else {
+			return isEqual(this.resource, EditorResourceAccessor.getCanonicalUri(otherInput)) && this.editorId === (otherInput as IResourceEditorInputType).options?.override;
```

Comment:
Isn't this a dangerous check here? If an editor does not implement `editorId`, it will be `undefined`. And if an editor opens without `override`, this editor matches if the resource is the same. Should we require a `editorId` here to match properly?